fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/GetNamespacedProperties.java

```diff
@@ -54,11 +57,21 @@ public FedoraEvent apply(final FedoraEvent evt) {
             final String[] parts = property.split(":", 2);
             if (parts.length == 2) {
                 final String prefix = parts[0];
-                try {
-                    event.addProperty(namespaceRegistry.getURI(prefix) + parts[1]);
-                } catch (RepositoryException ex) {
-                    LOGGER.trace("Prefix could not be dereferenced using the namespace registry: {}", property);
-                    event.addProperty(property);
+                if ("jcr".equals(prefix)) {
+                    if (jcrProperties.contains(createProperty(REPOSITORY_NAMESPACE + parts[1]))) {
+                        event.addProperty(REPOSITORY_NAMESPACE + parts[1]);
+                    } else if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug("Swallowing jcr property: {}", property);
+                    }
+                } else {
+                    try {
+                        event.addProperty(namespaceRegistry.getURI(prefix) + parts[1]);
+                    } catch (RepositoryException ex) {
+                        if (LOGGER.isDebugEnabled()) {
```

Comment:
Please remove the `LOGGER.isDebugEnabled()` guard to make it easier to change the logging state at runtime.